io/pci/pci_hotplug.py

```diff
@@ -47,76 +47,88 @@ def setUp(self):
             self.cancel("Test Unsupported! on this platform")
         if cpu._list_matches(open('/proc/cpuinfo').readlines(),
                              'platform\t: pSeries\n'):
-            power_vm = True
+            self.power_vm = True
             for mdl in ['rpaphp', 'rpadlpar_io']:
                 if not linux_modules.module_is_loaded(mdl):
                     linux_modules.load_module(mdl)
         elif cpu._list_matches(open('/proc/cpuinfo').readlines(),
                                'platform\t: PowerNV\n'):
-            power_vm = False
+            self.power_vm = False
             if not linux_modules.module_is_loaded("pnv_php"):
                 linux_modules.load_module("pnv_php")
         self.return_code = 0
+        self.err_pci = []
         self.device = self.params.get('pci_device', default=' ')
-        self.num_of_hotplug = int(self.params.get('num_of_hotplug', default='1'))
-        if not os.path.isdir('/sys/bus/pci/devices/%s' % self.device):
-            self.cancel("PCI device given does not exist")
-        self.num_of_hotplug = int(self.params.get('num_of_hotplug', default='1'))
-        if power_vm:
+        self.count = int(self.params.get('count', default='1'))
+        if not self.device:
+            self.cancel("PCI_address not given")
+
+    def get_slot(self, pci_addr):
+        '''
+        Returns the slot number with pci_address
+        '''
+        if self.power_vm:
             devspec = genio.read_file("/sys/bus/pci/devices/%s/devspec"
-                                      % self.device)
-            self.slot = genio.read_file("/proc/device-tree/%s/ibm,loc-code"
-                                        % devspec)
-            self.slot = re.match(r'((\w+)[\.])+(\w+)-P(\d+)-C(\d+)|Slot(\d+)',
-                                 self.slot).group()
+                                      % pci_addr)
+            slot = genio.read_file("/proc/device-tree/%s/ibm,loc-code"
+                                   % devspec)
+            slot = re.match(r'((\w+)[\.])+(\w+)-P(\d+)-C(\d+)|Slot(\d+)',
+                            slot).group()
         else:
-            self.slot = pci.get_pci_prop(self.device, "PhySlot")
-        if not os.path.isdir('/sys/bus/pci/slots/%s' % self.slot):
-            self.cancel("%s Slot not available" % self.slot)
-        if not os.path.exists('/sys/bus/pci/slots/%s/power' % self.slot):
-            self.cancel("%s Slot does not support hotplug" % self.slot)
+            slot = pci.get_pci_prop(pci_addr, "PhySlot")
+        if not os.path.isdir('/sys/bus/pci/slots/%s' % slot):
+            self.fail("%s Slot not available" % slot)
+        if not os.path.exists('/sys/bus/pci/slots/%s/power' % slot):
+            self.fail("%s Slot does not support hotplug" % slot)
+        return slot
```

Comment:
@Naresh-ibm I thought we concluded on this being called from setUp() and `cancel`ing these. Also, that was how it was in older method. Reason for change ?

Reply:
@narasimhan-v 

 1. Ideally, setUp() functions called from setUp() should cancel, and test() and functions called
     from test() should fail.
 2.  Here, if you want to cancel, call it from setUp()


with reference to your comment, since I am calling from test(), I am failing it.
previously I have used CANCEL only in the test(). then you gave the above review. hence I have changed it.

Reply:
```
Here, if you want to cancel, call it from setUp()
```